### PR TITLE
Moving Base64UrlEncode/Base64UrlDecode to separate entity

### PR DESCRIPTION
**Pull request by abatishchev**
- Adding IBase64UrlEncoder implemented by JwtBase64UrlEncoder. Calling it from JsonWebToken.Base64UrlEncode/Base64UrlDecode.
- Updating tests

### CODE REVIEW

**Comment by abatishchev** [comment]
I'll wait CI/CD to run tests automatically first

File: package.cmd [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-
+﻿tools\nuget.exe update -self
+
+if not exist package mkdir package
+if not exist package mkdir package
+if not exist package\lib mkdir package\lib
+if not exist package\lib\net35 mkdir package\lib\net35
+
+msbuild src\JWT.sln -p:Configuration=Release
+copy src\JWT\bin\Release\JWT.dll package\lib\net35
```
Comment (nbarbettini) [inline package.cmd]: Looks like some extra changes got moved into this PR?

File: src/JWT/IBase64UrlEncoder.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace JWT
+{
+    public interface IBase64UrlEncoder
+    {
+        string UrlEncode(byte[] input);
```
Comment (nbarbettini) [inline src/JWT/IBase64UrlEncoder.cs]: Seems a bit redundant to have `Url` twice, perhaps the method could just be `Encode` and `Decode`?

Comment (abatishchev) [inline src/JWT/IBase64UrlEncoder.cs]: Yeah, was thinking about this too. Drop the prefix.

**Pull request merged**